mmhuman3d/models/body_models/star.py

```diff
@@ -0,0 +1,212 @@
+             v.J_transformed:batch size x 24 x 3
+                            Posed model joints.
+             v.f: A numpy array of the model face.
+        """
+        device = pose.device
+        batch_size = pose.shape[0]
+        v_template = self.v_template[None, :]
+        shapedirs = self.shapedirs.view(-1, self.num_betas)[None, :].expand(
+            batch_size, -1, -1)
+        beta = betas[:, :, None]
+        v_shaped = torch.matmul(shapedirs, beta).view(-1, 6890, 3) + v_template
+        J = torch.einsum('bik,ji->bjk', [v_shaped, self.J_regressor])
+
+        pose_quat = quat_feat(pose.view(-1, 3)).view(batch_size, -1)
+        pose_feat = torch.cat((pose_quat[:, 4:], beta[:, 1]), 1)
+
+        R = rodrigues(pose.view(-1, 3)).view(batch_size, 24, 3, 3)
+        R = R.view(batch_size, 24, 3, 3)
+
+        posedirs = self.posedirs[None, :].expand(batch_size, -1, -1)
+        v_posed = v_shaped + torch.matmul(
+            posedirs, pose_feat[:, :, None]).view(-1, 6890, 3)
+
+        J_ = J.clone()
+        J_[:, 1:, :] = J[:, 1:, :] - J[:, self.parent, :]
+        G_ = torch.cat([R, J_[:, :, :, None]], dim=-1)
+        pad_row = torch.FloatTensor([0, 0, 0,
+                                     1]).to(device).view(1, 1, 1, 4).expand(
+                                         batch_size, 24, -1, -1)
+        G_ = torch.cat([G_, pad_row], dim=2)
+        G = [G_[:, 0].clone()]
+        for i in range(1, 24):
+            G.append(torch.matmul(G[self.parent[i - 1]], G_[:, i, :, :]))
+        G = torch.stack(G, dim=1)
+
+        rest = torch.cat([J, torch.zeros(batch_size, 24, 1).to(device)],
+                         dim=2).view(batch_size, 24, 4, 1)
+        zeros = torch.zeros(batch_size, 24, 4, 3).to(device)
+        rest = torch.cat([zeros, rest], dim=-1)
+        rest = torch.matmul(G, rest)
+        G = G - rest
+        T = torch.matmul(self.weights,
+                         G.permute(1, 0, 2, 3).contiguous().view(24, -1)).view(
+                             6890, batch_size, 4, 4).transpose(0, 1)
+        rest_shape_h = torch.cat(
+            [v_posed, torch.ones_like(v_posed)[:, :, [0]]], dim=-1)
+        v = torch.matmul(T, rest_shape_h[:, :, :, None])[:, :, :3, 0]
+        v = v + trans[:, None, :]
+        v.f = self.f
+        v.v_posed = v_posed
+        v.v_shaped = v_shaped
+
+        root_transform = with_zeros(
+            torch.cat((R[:, 0], J[:, 0][:, :, None]), 2))
+        results = [root_transform]
+        for i in range(0, self.parent.shape[0]):
+            transform_i = with_zeros(
+                torch.cat((R[:, i + 1], J[:, i + 1][:, :, None] -
+                           J[:, self.parent[i]][:, :, None]), 2))
+            curr_res = torch.matmul(results[self.parent[i]], transform_i)
+            results.append(curr_res)
+        results = torch.stack(results, dim=1)
+        posed_joints = results[:, :, :3, 3]
+        v.J_transformed = posed_joints + trans[:, None, :]
+        return v
```

Comment:
Is it possible to follow other body models and
1. make the output in the dictionary form
2. add the keypoint remapping

I have provide some discussion point [here](https://github.com/open-mmlab/mmhuman3d/discussions/162)。 Appreciate your feedback. ;)